### ISSUE TITLE
Exception when reading document with nullable int field

### ISSUE THREAD

**Issue by evsheffield** [scott-mcdonald/JsonApiFramework#32]
I'm getting an exception thrown when trying to read a document which has a `null` value set for a field which is a nullable int.

**Resource**

```
public class Node : IResource
{
    public Guid Id { get; set; }
    public int? RowCount { get; set; }
}
```

**Document to be read**

```
{
  "data": {
    "type": "nodes",
    "id": "6ee388c9-520b-e711-8e0b-f0d5bf603af2",
    "attributes": {
      "row-count": null
    }
  }
}
```

**Code to read document**

```
public void ReadDocument(Uri requestUri, string json)
{
    var document = JsonObject.Parse<Document>(json);
    using (var documentContext = new ReportingDocumentContext(requestUri, json))
    {
        var documentType = documentContext.GetDocumentType();
        Contract.Assume(documentType == DocumentType.ResourceDocument);
        var node = documentContext.GetResource<Node>();
    }
}
```

This generates an exception with the message `Unable to convert from JValue to Nullable`, and the following inner exception:

```
System.ArgumentException: Can not convert Null to Int32.\r\n   at Newtonsoft.Json.Linq.JToken.op_Explicit(JToken value)\r\n   at Newtonsoft.Json.Linq.JToken.ToObject(Type objectType)\r\n   at JsonApiFramework.Reflection.TypeConverter.HandleSpecialCaseTypesOfJTokenAndObject(Object sourceValue, Type targetType, Object& targetValue)\r\n   at JsonApiFramework.Reflection.TypeConverter.TryConvertInternal(Object sourceValue, Type targetType, IFormatProvider targetFormatProvider, Object& targetValue)\r\n   at JsonApiFramework.Reflection.TypeConverter.Convert(Object sourceValue, Type targetType, IFormatProvider targetFormatProvider)
```

I haven't had the chance to set up a test case for it yet, but based on a quick look through the code for `TypeConverter` my suspicion is that the source `JValue` fails the equality check for null [here](https://github.com/scott-mcdonald/JsonApiFramework/blob/d95378688e59d73914867ce926fa0c18b8017e91/Source/JsonApiFramework.Core/Reflection/TypeConverter.cs#L368) which then leads to the target type being stripped of the nullable component in [this block](https://github.com/scott-mcdonald/JsonApiFramework/blob/d95378688e59d73914867ce926fa0c18b8017e91/Source/JsonApiFramework.Core/Reflection/TypeConverter.cs#L401) which leads to the exception when it tries to convent null to a non-nullable int.

**Comment by scott-mcdonald** [comment]
@evsheffield FYI I have not forgotten about this defect. In middle of trying to get a release candidate for public facing product that is consuming a lot of time. I promise this will be fixed next week...

**Comment by evsheffield** [comment]
No worries, @scott-mcdonald, I definitely understand that feeling. Best of luck and thanks for your help!

**Comment by scott-mcdonald** [comment]
@evsheffield Was able to reproduce the defect you discovered with a unit test. Look for the following unit tests if you are interested:

> WithPosSystemAndNonNullEndOfLifeDateResourceDocument
> WithPosSystemAndNullEndOfLifeDateResourceDocument

Fixed the defect, give me a couple of days to build and publish the next version of the framework to nuget. This was a good catch, thanks for reporting it...

**Comment by scott-mcdonald** [comment]
Reopening until I publish latest version to nuget.

**Comment by scott-mcdonald** [comment]
@evsheffield Ok Evan, I published v1.2.1-beta of the framework which I believe will fix this issue. Going to close...

**Comment by evsheffield** [comment]
@scott-mcdonald Sorry for the delayed response, I just wanted to thank you for getting this fix pushed out so quickly. Everything's working as expected now. Much appreciated!
